.github/workflows/ci_test-full.yml

```diff
@@ -89,19 +89,19 @@ jobs:
       run: |
         flag=$(python -c "print('--pre' if '${{matrix.release}}' == 'pre' else '')" 2>&1)
         url=$(python -c "print('test/cpu/torch_test.html' if '${{matrix.release}}' == 'pre' else 'cpu/torch_stable.html')" 2>&1)
-        pip install -r requirements.txt --upgrade $flag --find-links "https://download.pytorch.org/whl/${url}"
+        pip install -r requirements.txt $flag -U --upgrade-strategy eager --find-links "https://download.pytorch.org/whl/${url}"
         # adjust versions according installed Torch version
         python ./requirements/adjust-versions.py requirements/examples.txt
-        pip install -r requirements/examples.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html --upgrade
-        pip install -r requirements/test.txt --upgrade
+        pip install -r requirements/examples.txt --find-links https://download.pytorch.org/whl/cpu/torch_stable.html -U --upgrade-strategy eager
+        pip install -r requirements/test.txt -U --upgrade-strategy eager
```

Comment:
CI is failing because a `test.txt` dependency requires `protobuf` and since we are marking eager here, it gets upgraded to the latest version since it's not considering the pin in `requirements.txt`

Not sure about how we can resolve this problem. All requirements would need to be installed together instead of in steps so the resolution algorithm can find a working combination.